Lib/pathlib.py

```diff
@@ -178,11 +178,11 @@ def _iterate_directories(self, parent_path, scandir):
             for entry in entries:
                 entry_is_dir = False
                 try:
-                    entry_is_dir = entry.is_dir()
+                    entry_is_dir = entry.is_dir(follow_symlinks=False)
                 except OSError as e:
                     if not _ignore_error(e):
                         raise
-                if entry_is_dir and not entry.is_symlink():
+                if entry_is_dir:
```

Comment:
Interesting, we intentionally don't follow symlinks in this glob() implementation?

Reply:
Yep! See #70200 where it was changed.